fe/src/main/java/org/apache/doris/catalog/ScalarType.java

```diff
@@ -141,6 +141,9 @@ public static ScalarType createType(PrimitiveType type) {
                 return DEFAULT_DECIMALV2;
             case LARGEINT:
                 return LARGEINT;
+            case ARRAY:
+                // can't static, because the same array may cause circular dependency
```

Comment:
What does this mean?

Reply:
Nested type need decide sub-element type in runtime, return a static object will take repeatedly set the sub-element type of same object in many places......

Reply:
Nested type need decide  and set sub-element type in runtime, return a static object will take repeatedly set the sub-element type of same object in many places......

Reply:
Nested type need decide sub-element type in runtime, return a static object will take repeatedly set the sub-element type of same object in many places......